src/auxiliary/mpi_arrays.jl

```diff
@@ -0,0 +1,332 @@
+
+# TODO: MPI. Keep this module inside Trixi or move it to another repo as
+#            external dependency with simple test suite and documentation?
```

Comment:
My initial thought: depends on our goals. If we really only want the minimum functionality to support our use case, I don't think having a separate package is worth the hassle.

If, on the other hand, we see the benefit for others (and us) if also other people use it, we should make it a separate package.

If we just want to give it a try and see if anyone is interested in chiming in, we should probably release it as a separate package but then also consider to use a more neutral name.

Reply:
I think it's fine to keep it in Trixi.jl for now, develop and test it further, and decide after a faw weeks after finishing this PR.

Reply:
Sounds good!